### PR TITLE
Fix deprecations and remove Bayesian models

### PR DESCRIPTION
**Pull request by twiecki**
This fixes a whole bunch of stuff that caused tests to break under more recent python and pandas. It also removes the bayesian module, it's just making tests slow, is difficult to install, and we don't have time to maintain it properly.

### CODE REVIEW

**Comment by twiecki** [comment]
@huaiweicheng Getting closer, still a few legit unit test failures. Want to give fixing those a try?

**Comment by huaiweicheng** [comment]
@twiecki  Yeah, let me compelete the whole thing.

**Comment by twiecki** [comment]
@huaiweicheng Let me know if you need help.

**Comment by huaiweicheng** [comment]
@twiecki  Some other work cost me some time. Will continue in near days.

File: pyfolio/tests/test_round_trips.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ class RoundTripTestCase(TestCase):
                          ],
                    columns=['amount', 'price', 'symbol'],
                    index=dates_intraday[[0, 2]])
-         .rename_axis('dt', axis='index')
+         .rename('dt', axis='index')
```
Comment (huaiweicheng) [inline pyfolio/tests/test_round_trips.py]: This might be wrong. revert it.

File: pyfolio/tests/test_round_trips.py [hunk 2]
```diff
@@ -53,7 +53,7 @@ class RoundTripTestCase(TestCase):
                          ],
                    columns=['amount', 'price', 'symbol'],
                    index=dates_intraday[[0, 4]])
-         .rename_axis('dt', axis='index')
+         .rename('dt', axis='index')
```
Comment (huaiweicheng) [inline pyfolio/tests/test_round_trips.py]: This might be wrong. revert it.

**Comment by huaiweicheng** [comment]
@twiecki  Reverting the above codes and I have passed all the test right now.  Did you have any suggestions? (fyi, never used round_trip before and not familiar with that part)

**Comment by twiecki** [comment]
@huaiweicheng Want to do another PR with that fix?

**Comment by huaiweicheng** [comment]
@twiecki Sorry to get you late. Of course I'll do.

**Comment by huaiweicheng** [comment]
@twiecki  python3.7 works. python2.7 did not pass. I notice the failed reason is that the difference of nan & 0. Could you help locating that?

**Comment by huaiweicheng** [comment]
@twiecki  I find another problem. The 

```
    for column in perf_stats.columns:
        for stat, value in perf_stats[column].iteritems():
            if stat in STAT_FUNCS_PCT:
                perf_stats.loc[stat, column] = str(np.round(value * 100,
                                                            1)) + '%
```

code  in line 650 of plotting.py makes the precision to 1 which is to small.  My suggestion is that precision should be at least 2 or 3.

**Comment by twiecki** [comment]
We did it @huaiweicheng!! Thanks so much for your help.

**Comment by twiecki** [comment]
@richafrank Probably should have looped you in before merging, but this fixes a bunch of stuff and removes a bunch of stuff which we don't need anymore. Mainly the Bayes tearsheet and the risk tearsheet (superseded by the perf_attrib tearsheet).

**Comment by twiecki** [comment]
Thanks, fixed on master.

**Comment by huaiweicheng** [comment]
Hi, twiecki. How about adding turnover_denom para in create_txn_tear_sheet function? turnover_denom is one para in create_returns_tear_sheet but not in create_txn_tear_sheet which is weird. I could do the PR.

**Comment by twiecki** [comment]
Is this a feature you need? But yes, I'd probably merge a PR with that,
thanks for offering.

**Comment by huaiweicheng** [comment]
@twiecki  Yeah. I use the create_returns_tear_sheet  to produce a brief summary data of returns, ratios, turnovers and so on. The turnover_denom param I set it to  'portfolio_value'. Meanwhile I use create_txn_tear_sheet to generate the report, I find it has no turnover_denom param. Thus, the turnover data generated by create_returns_tear_sheet  and the plot generated by create_txn_tear_sheet does not match. I open a new PR #618, which locate the problem. And also, it fix some small issues such as useless params.

**Comment by steelep** [comment]
> This fixes a whole bunch of stuff that caused tests to break under more recent python and pandas. It also removes the bayesian module, it's just making tests slow, is difficult to install, and we don't have time to maintain it properly.

If I spent some time to reincorporate the bayesian module - would you be open to this? It was super useful.

**Pull request merged**